ChainSafe/web3.unity
# Marketplace sample calls

Marketplace write calls are now functional within the sample scene, these functions can easily be added to a GUI to populate it with data as the user wishes.

![image](https://github.com/ChainSafe/web3.unity/assets/57473220/fad30374-80c5-4de2-8e5d-252129a2d601)

To test:
- You will need a bearer token from the dashboard login console https://dashboard.gaming.chainsafe.io/ it comes from the login response in the network tab and its member key is access_token, we need a better way to expose this to the user as it's needed for the request header with half of the calls.
- You will also need to create a 721, 1155 collection with Nfts and a marketplace to see the get calls, you can also do this from the editor if you choose.
- Insert all the details into the Marketplace Call object and fill it out with contract & id information. Be careful with IDs and contracts, they're two different things.
- As the contracts are specific to users and have role control, they need to be initialized by the person using them for functions like mint which is why i have left the variables open.
- Press play and test the marketplace functions in the sample scene.
- Newly created collections, marketplaces and minted Nfts will show up accordingly in the dashboard.
- Deleting may be sticky to test as it can only be used after chain interactions are reverted for creating collections/marketplaces.
- Server settings menu moved up a level in the editor as it's more intuitive there.
